lib/base/VariableResolver.js

```diff
@@ -309,9 +309,9 @@ function merge(property, target, source) {
     target[property] = [];
   }
 
-  const originToAdd = source[property].filter(o => !target.origin.includes(o));
+  const propertiesToAdd = source[property].filter(o => !target[property].includes(o));
 
-  target[property].push(...originToAdd);
+  target[property].push(...propertiesToAdd);
```

Comment:
If spread can cause problems, why do we keep using it?

```suggestion
  target[property] = target[property].concat(propertiesToAdd);
```